docs/DeveloperGuide.adoc

```diff
@@ -734,16 +776,138 @@ Use case ends.
 +
 Use case resumes at step 2.
 
-{More to be added}
+
+[discrete]
+=== Use case: Add person
+
+*MSS*
+
+1.  User requests to add person with the person’s name, phone number and email address
+2.  AddressBook adds the person
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. User did not give enough details
++
+AddressBook displays an error message
++
+Use case ends.
+
+
+[discrete]
+=== Use case: View command history
+
+*MSS*
+
+1.  User requests to view command history
+2.  AddressBook displays the command history in reverse chronological order
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. AddressBook displays an empty list
++
+Use case ends.
+
+
+[discrete]
+=== Use case: find a person in the address book
+
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User inputs the name of the person
+4.  AddressBook returns the people with a similar name and their details
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. AddressBook displays an empty list
++
+Use case ends.
+
+* 4a. No person is found
++
+[none]
+** 3a1. AddressBook shows an empty list
++
+Use case ends.
+
+
+[discrete]
+=== Use case: Edit person
+
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User requests to edit a specific person in the list with new details
+4. AddressBook updates the person’s details
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. AddressBook displays an empty list
++
+Use case ends.
+
+[none]
+* 3a. User did not give enough details
++
+** 4a1. AddressBook displays an error message
++
+Use case ends.
+
+
+[discrete]
+=== Use case: Add a person from their social media account
+
+*MSS*
+
+1.  User requests to add a Facebook or Linkedin URL
+2.  AddressBook adds the person
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. Given URL is broken
++
+** 2a1. AddressBook displays an error message.
++
+Use case ends.
+
+[none]
+* 1b. Given URL is from Facebook
++
+** 2a1. AddressBook asks the user to input missing details.
++
+Use case resumes at step 2.
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any link:#mainstream-os[mainstream OS] as long as it has Java `1.8.0_60` or higher installed.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Retrieve the correct persons the user searches using specific keywords
+.  Should back up the database to more than one location
+.  Extract only relevant user information when adding a person from the web
+.  Use an link:#open-standard[open standard] to store the user data to allow for editing outside of the address book and easy debugging
```

Comment:
This sounds more like a user story?? since you're implementing something into addressbook. 
you can probably word it better to make it sound more like a NFR

Same goes to `Extract only relevant user information when adding a person from the web`